File: frontend/src/services/followService.js
import api from './api';

const followService = {
  toggleFollow: async (targetId) => {
    const response = await api.post(`/follows/${targetId}/toggle`);
    return response;
  },

  getFollowing: async () => {
    const response = await api.get('/follows/following');
    return response;
  },

  getFollowers: async (targetId) => {
    const response = await api.get(`/follows/followers/${targetId}`);
    return response;
  },

  checkFollowStatus: async (targetId) => {
    const response = await api.get(`/follows/status/${targetId}`);
    return response;
  }
};

export default followService;

File: backend/src/main/java/vn/kurisu/mentormatch/service/FollowService.java
package vn.kurisu.mentormatch.service;

import vn.kurisu.mentormatch.dto.response.ApiResponse;
import vn.kurisu.mentormatch.dto.response.UserResponse;

import java.util.List;

public interface FollowService {
    ApiResponse<String> toggleFollow(Integer targetUserId);
    ApiResponse<List<UserResponse>> getFollowing();
    ApiResponse<List<UserResponse>> getFollowers(Integer targetUserId);
    ApiResponse<Boolean> checkFollowStatus(Integer targetUserId);
}
